pallets/identity/src/lib.rs

```diff
@@ -2141,6 +2141,47 @@ impl<T: Trait> Module<T> {
 
         Self::deposit_event(RawEvent::DidCreated(id, acc, vec![]));
     }
+
+    /// It returns the list of flatten identities of the given identity.
+    /// It runs recursively over all signing items.
+    pub fn flatten_identities(id: IdentityId) -> Vec<IdentityId> {
+        if <DidRecords>::contains_key(id) {
+            let identity = <DidRecords>::get(id);
+
+            identity
+                .signing_items
+                .into_iter()
+                .flat_map(|si| match si.signer {
+                    Signatory::Identity(sub_id) => Self::flatten_identities(sub_id),
+                    _ => vec![],
+                })
+                .chain([id].iter().cloned())
+                .collect::<Vec<_>>()
+        } else {
+            vec![]
+        }
+    }
+
+    /// Get the list of flatten keys fo the given identity.
+    /// It runs recursively over all signing items.
+    pub fn flatten_keys(id: IdentityId) -> Vec<AccountKey> {
```

Comment:
I think having the flatten functions is fine (they might come handy in future) but we don't need to use them in current usecase.

Reply:
I've kept these functions, adding a new parameter to limit the maximun depth for recursion.
In the historical voting RPC, we use `max_depth == 1` to use the non-recursive approach.